dev/tools/merge-pr.sh

```diff
@@ -1,50 +1,142 @@
 #!/usr/bin/env bash
 
 set -e
+set -o pipefail
+
+API=https://api.github.com/repos/coq/coq
 
 # This script depends (at least) on git and jq.
 # It should be used like this: dev/tools/merge-pr.sh /PR number/
 
-#TODO: check arguments and show usage if relevant
+RED="\033[31m"
+BLACK="\033[30m"
+GREEN="\033[32m"
+BLUE="\033[34m"
+YELLOW="\033[33m"
+info() {
+  echo -e "${GREEN}info:${BLACK} $1 ${BLACK}"
+}
+error() {
+  echo -e "${RED}error:${BLACK} $1 ${BLACK}"
+}
+warning() {
+  echo -e "${YELLOW}warning:${BLACK} $1 ${BLACK}"
+}
 
-PR=$1
+check_util() {
+if ! which "$1" > /dev/null; then
```

Comment:
Don't you want to `2 > &1`? Have you tested with a missing util?

Reply:
>Don't you want to `2 > &1`? Have you tested with a missing util?

After you said that I looked up which specs (they don't exist, which isn't posix) and it seems macos which exits with 0 even for missing utilities https://stackoverflow.com/questions/762631/find-out-if-a-command-exists-on-posix-system
TLDR use `if ! command -v "$1" > /dev/null`

Reply:
Does it mean that the following line is also broken? https://github.com/coq/coq/blob/ef0fc7730df3cb585b621955010115407d80d2e7/Makefile.build#L211

Reply:
posix epic fail:
```
$ gpg -v
gpg: WARNING: no command supplied.  Trying to guess what you mean ...
gpg: Go ahead and type your message ...
^C
gpg: signal Interrupt caught ... exiting
```

Reply:
maybe we should try with:
```
echo masochistic UI | $command -v
```

Can you dismiss this change request?

Reply:
No I mean the builtin `command`

ie 
```sh
$ command -v bash
/bin/bash (or wherever your bash is)
$ echo $?
0
$ command -v foobarbaz
$ echo $?
1
```

Reply:
FWIW, on my OS X,  both buit-in csh `which` and `/usr/bin/which` returns 1 if the command does not exist.

Reply:
thanks I'll fix that